### PR TITLE
fix: (platform) split menu button fix for label change on action selection from menu

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/3086

#### Please provide a brief summary of this pull request.
this PR fixes issues that were raised during defect hunting.
1. on action option select from menu, default button action gets updated.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by DeepakSap14** [comment]
before:
broken-platform-split
After:
split-menu

**Review by KevinOkamoto — COMMENTED** [review]
The menu still looks like it's dropping to the right.

https://deploy-preview-3230--fundamental-ngx.netlify.app/#/platform/split-menu-button

Screen Shot 2020-09-10 at 7 13 05 AM

**Comment by DeepakSap14** [comment]
> The menu still looks like it's dropping to the right.
> 
> https://deploy-preview-3230--fundamental-ngx.netlify.app/#/platform/split-menu-button
> 
> Screen Shot 2020-09-10 at 7 13 05 AM

fdpMenuTriggerFor directive is applied on second button of split-menu-button. so it is taking position position relative to second button.
currently fdpMenuTriggerFor, does not give option to position menu overlay.

**Comment by KevinOkamoto** [comment]
You can position the dropdown by setting the `xPosition` on the MenuComponent:

https://github.com/SAP/fundamental-ngx/blob/a1f32de89dd3fbceda08e56b8a81206b6f4c0bfd/libs/platform/src/lib/components/menu/menu.component.ts#L71-L75

**Review by KevinOkamoto — CHANGES_REQUESTED** [review]
Try adding `xPosition="before"` to the fdp-menu components.

**Comment by DeepakSap14** [comment]
fixed now.
Screenshot 2020-09-11 at 9 11 37 AM

Screenshot 2020-09-11 at 9 11 27 AM

**Review by KevinOkamoto — APPROVED** [review]

**Comment by stefanoScalzo** [comment]
You might want to wait on https://github.com/SAP/fundamental-ngx/pull/3239 for this

**Comment by DeepakSap14** [comment]
PR review is in progress.

File: apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html [hunk 1]
```diff
@@ -1,98 +1,197 @@
...
-    [icon]="'world'"
-    i18n-buttonLabel
-    buttonLabel="Default Action"
-    i18n-title
-    title="Default action button with list of other available action items"
-    (primaryButtonClick)="onPrimaryButtonClickCompact()"
-    i18n="@@basicSplitMenuBtn"
->
+<h3>Internationalization support with i18n directive</h3>
+<fdp-split-menu-button id="splitButton14" [menu]="basicMenuIntz" icon="world" i18n-buttonLabel [buttonLabel]="label"
```
Comment (dimamarksman) [inline apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html:116]: Do we really need to assign all these `ids`, as for me it bloats examples and doesn't bring any value.

File: apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html [hunk 2]
```diff
@@ -1,98 +1,197 @@
...
+    <fdp-menu-item (itemSelect)="onItemSelectCompactTransparent('Transparent Button 3')">Transparent Button 3
+    </fdp-menu-item>
+</fdp-menu>
+<fdp-menu #basicMenuCompactEmphasized id="basic-menu-Compact-emphasized" xPosition="before">
+    <fdp-menu-item (itemSelect)="onItemSelectCompactEmphasized('Emphasized Button')">Emphasized Button</fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactEmphasized('Emphasized Button 2')">Emphasized Button 2
+    </fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactEmphasized('Emphasized Button 3')">Emphasized Button 3
+    </fdp-menu-item>
 </fdp-menu>
```
Comment (dimamarksman) [inline apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html:197]: I believe we should avoid examples with 200 lines of code, that's not readable at all, I just got lost there in one minute :)

If we want to show available `button types` we should show just it. We shouldn't repeat it twice for `comact` mode if it was already shown in `cozy` mode cause it does not bring any additional value to user.

Could we reuse one `fdp-menu` for all buttons to reduce the example code.

**Review by dimamarksman — COMMENTED** [review]
More questions

File: apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html (hunk 1 above)
Comment (DeepakSap14) [reply apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html]: removed ids.

File: apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html (hunk 2 above)
Comment (DeepakSap14) [reply apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html:206]: Hi dima,
i can split the file to have examples of cozy and compact different.
but for meeting testing team requirement, i have to include examples for all states and for cozy-compact both.

for having a single "fdp-menu":
in current implementation i am calling method from menu-item selection, and this method updates the label for primary button.
if i keep a single fdp-menu, then selecting a item in one split-menu-button, will change label for all other split-menu-button as well. which will be wrong. so i have to keep separate fdp-menu.

I am raising a separate PR, where feature for updating the label for primary button will be moved inside component.
As this PR was part of defect hunting, i will not be doing this change in this PR.

**Comment by DeepakSap14** [comment]
> You might want to wait on #3239 for this

I am raising a separate PR, where feature for updating the label for primary button will be moved inside component.
As this PR was part of defect hunting, i will not be doing this change in this PR.

File: apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html (hunk 2 above)
Comment (dimamarksman) [reply apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html:206]: I see your point, sounds good

**Review by Betrozov — APPROVED** [review]

**Review by dimamarksman — APPROVED** [review]

File: apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html (hunk 2 above)
Comment (dimamarksman) [reply apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html:206]: @DeepakSap14 

> but for meeting testing team requirement, i have to include examples for all states and for cozy-compact both.

If create these examples for end users (app developers) we should not adapt it for our testing team.
I believe testing team should have additional tools to test it properly.

**Comment by mikerodonnell89** [comment]
The specs state that when selecting a menu option that is of a different length than the initially selected option, the button should remain the same size and the text should truncate with ellipses.

I've been working on this component over the past few days in ngx and it may be useful to see the proposed changes here:
https://github.com/SAP/fundamental-ngx/pull/3239

**Comment by DeepakSap14** [comment]
I have added truncation behaviour and included examples.
I have distributed examples between files, for easy review.
please review.

**Comment by DeepakSap14** [comment]
@dimamarksman i have talked with testing team, and then changed examples, the way you were suggesting. you can review it.

**Comment by dimamarksman** [comment]
> @dimamarksman i have talked with testing team, and then changed examples, the way you were suggesting. you can review it.

Thanks, now it's much better!

**Review by InnaAtanasova — APPROVED** [review]

**Review by KevinOkamoto — APPROVED** [review]
Looks good.

**Pull request merged**